### PR TITLE
[DowngradePhp80] Support `match` as array item in DowngradeMatchToSwitchRector

### PR DESCRIPTION
**Pull request by jrstanley**
Currently exploring how to add support for this.

I have added three failing test fixtures, however I am not certain that the 'expected' result of these is correct.

I am confident that the current behaviour of the `DowngradeMatchToSwitchRector` rule is incorrect when treating `match` in the context of arrays.

### CODE REVIEW

**Comment by jrstanley** [comment]
@TomasVotruba / @samsonasik I think that this is ready for your kind review

**Comment by samsonasik** [comment]
@jrstanley could you rebase/amend last commit to remove quoted 'main' in commit message? That can make build stopped when merged.

File: rules/DowngradePhp80/Rector/Expression/DowngradeMatchToSwitchRector.php [hunk 1]
```diff
@@ -84,29 +92,45 @@ public function run()
...
+            $node->value = new FuncCall($this->anonymousFunctionFactory->create([], [$switch], null));
+            return $node;
+        }
+
+        return $switch;
+    }
+
+    private function shouldSkipNode(Node $node): bool
+    {
+        if ($node instanceof Return_ && ! $node->expr instanceof Match_) {
```
Comment (samsonasik) [inline rules/DowngradePhp80/Rector/Expression/DowngradeMatchToSwitchRector.php:126]: You can directly return condition for true else false here.

```
return $node instanceof Return_ && ! $node->expr instanceof Match_;

**Comment by jrstanley** [comment]
> @jrstanley could you rebase/amend last commit to remove quoted 'main' in commit message? That can make build stopped when merged.

If I push a new commit to address your review comment, will this be sufficient? Or do I need to remove the reference to `main` from the branch history?

**Comment by samsonasik** [comment]
It only need quote removal from commit message, a rebase can clean up that:

```
Merge branch 'main' of
```

we can remove that on merge of course, just to avoid accidental merge without cleaned it up.

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
I will merge it and clean up commit at merge, the if else check clean up https://github.com/rectorphp/rector-src/pull/2178#discussion_r862393272  should can happen by next rectify.

**Comment by samsonasik** [comment]
Thank you @jrstanley

**Pull request merged**